components/core/src/clp_s/JsonConstructor.cpp

```diff
@@ -38,47 +36,20 @@ JsonConstructor::JsonConstructor(JsonConstructorOption const& option)
                 fmt::format("'{}' is not a directory", m_archives_dir)
         );
     }
-
-    for (auto const& entry : std::filesystem::directory_iterator(m_archives_dir)) {
-        if (false == entry.is_directory()) {
-            // Skip non-directories
-            continue;
-        }
-
-        m_archive_paths.push_back(entry.path().string());
-    }
-
-    if (m_archive_paths.empty()) {
-        throw OperationFailed(
-                ErrorCodeFailure,
-                __FILENAME__,
-                __LINE__,
-                fmt::format("No sub-archives in '{}'", m_archives_dir)
-        );
-    }
-
-    m_max_archive_index = m_archive_paths.size() - 1;
 }
 
 void JsonConstructor::construct() {
-    m_schema_tree = ReaderUtils::read_schema_tree(m_archives_dir);
-    auto id_to_schema = ReaderUtils::read_schemas(m_archives_dir);
-    auto timestamp_dict = ReaderUtils::read_timestamp_dictionary(m_archives_dir);
-
-    m_archive_reader = std::make_unique<ArchiveReader>(m_schema_tree, id_to_schema, timestamp_dict);
+    m_archive_reader = std::make_unique<ArchiveReader>();
```

Comment:
Judging by the docstrings this method should also at least open the archive and read schema metadata. Granted the docstrings for construct/store could also do with a rewrite, so maybe we can just change the description instead.

Reply:
Do you think we should make construct and store a single method?

Reply:
Yeah might as well